target/linux/ipq40xx/files-4.14/arch/arm/boot/dts/qcom-ipq4018-ens620ext.dts

```diff
@@ -0,0 +1,277 @@
+};
+
+&blsp_dma {
+	status = "okay";
+};
+
+&tlmm {
+	serial_pins: serial_pinmux {
+		mux {
+			pins = "gpio60", "gpio61";
+			function = "blsp_uart0";
+			bias-disable;
+		};
+	};
+
+	spi_0_pins: spi_0_pinmux {
+		mux {
+			function = "blsp_spi0";
+			pins = "gpio55", "gpio56", "gpio57";
+			drive-strength = <12>;
+			bias-disable;
+		};
+
+		mux_cs {
+			function = "gpio";
+			pins = "gpio54";
+			drive-strength = <2>;
+			bias-disable;
+			output-high;
+		};
+	};
+};
+
+&blsp1_spi1 { /* BLSP1 QUP1 */
+	pinctrl-0 = <&spi_0_pins>;
+	pinctrl-names = "default";
+	status = "okay";
+	cs-gpios = <&tlmm 54 GPIO_ACTIVE_HIGH>;
+
+	flash@0 {
+		#address-cells = <1>;
+		#size-cells = <0>;
+		/*
+		 * U-boot looks for "n25q128a11" node,
+		 * if we don't have it, it will spit out the following warning:
+		 * "ipq: fdt fixup unable to find compatible node".
+		 */
+		compatible = "jedec,spi-nor";
+		reg = <0>;
+		linux,modalias = "m25p80", "mx25l1606e", "n25q128a11";
+		spi-max-frequency = <24000000>;
+
+		partitions {
+			compatible = "fixed-partitions";
+			#address-cells = <1>;
+			#size-cells = <1>;
+
+			partition0@0 {
+				label = "0:SBL1";
+				reg = <0x00000000 0x00040000>;
+				read-only;
+			};
+			partition1@40000 {
+				label = "0:MIBIB";
+				reg = <0x00040000 0x00020000>;
+				read-only;
+			};
+			partition2@60000 {
+				label = "0:QSEE";
+				reg = <0x00060000 0x00060000>;
+				read-only;
+			};
+			partition3@c0000 {
+				label = "0:CDT";
+				reg = <0x000c0000 0x00010000>;
+				read-only;
+			};
+			partition4@d0000 {
+				label = "0:DDRPARAMS";
+				reg = <0x000d0000 0x00010000>;
+				read-only;
+			};
+			partition5@e0000 {
+				label = "0:APPSBLENV"; /* uboot env*/
+				reg = <0x000e0000 0x00010000>;
+				read-only;
+			};
+			partition5@f0000 {
+				label = "0:APPSBL"; /* uboot */
+				reg = <0x000f0000 0x00090000>;
+				read-only;
+			};
+			partition5@180000 {
+				label = "0:ART";
+				reg = <0x00180000 0x00010000>;
+				read-only;
+			};
+			partition5@190000 {
+				label = "0:HLOS";
+				reg = <0x00190000 0x00500000>;
+			};
+			partition5@690000 {
+				label = "rootfs";
+				reg = <0x00690000 0x00FD0000>;
+			};
+			partition5@1660000 {
+				label = "failsafe";
+				reg = <0x01660000 0x008F0000>;
+				read-only;
+			};
+			partition5@1f50000 {
+				label = "u-boot-env";
+				reg = <0x01F50000 0x00010000>;
+				read-only;
+			};
+			partition5@1f60000 {
+				label = "userconfig";
+				reg = <0x01F60000 0x000A0000>;
+				read-only;
+			};
+			partition9@190000 {
+				compatible = "denx,fit";
+				label = "firmware";
+				reg = <0x00190000 0x14D0000>;
+			};
+	/* 0x00180000 - 0x00200000 unused */
+		};
+	};
+
+};
+
+&blsp1_uart1 {
+	pinctrl-0 = <&serial_pins>;
+	pinctrl-names = "default";
+	status = "okay";
+};
+
+&usb3_ss_phy {
+	status = "okay";
+};
+
+&usb3_hs_phy {
+	status = "okay";
+};
+
+&usb2_hs_phy {
+	status = "okay";
+};
+
+&wifi0 {
+	status = "okay";
+	qcom,ath10k-calibration-variant = "EnGenius-ENS620EXT";
```

Comment:
You'll need to submit your device's boarddata files (see in the vendor firmware for files like boardData_0.bin under /lib/firmware/...) upstream to ath10k-devel mailing list.

You can find instructions and examples on the linux-wireless wiki:
https://wireless.wiki.kernel.org/en/users/drivers/ath10k/boardfiles

Once you are done, please pack the board-files into a ipq-wifi package for your device.
You can find instructions for how to do this in the ipq-wifi commit: fa03d441e96eeec0781ccbe757a3641c9dcec785

If you have any more questions regarding the boardfiles or ipq-wifi please ask.

Reply:
Thank you thank you thank you. Was not aware of how this worked. I built the board files with the qca swiss army knife and now 5GHz works properly!!! 

I'll submit the board files and the short term addition to ipq_wifi will be in the next commit.
Thanks for all the pointers - just what I needed.

Reply:
Thank you for posting the boardfiles to the ath10k mailing list.

http://lists.infradead.org/pipermail/ath10k/2019-March/013034.html